### PR TITLE
Actions docs

### PR DESCRIPTION
**Pull request by jennyfolkesson**
I decided to go with the following for creating a github actions workflow for autopublishing docs:
https://github.com/peaceiris/actions-gh-pages

In my last push I updated the action to only trigger when there's a push to main (which I haven't been able to test), but I've been testing it with this branch and it seems to be working, so hopefully this will work.

I've also added some notes in the docs/readme.md on how to build the html locally and with the workflow.

I've set up a branch named gh-pages where changes in documentation will be pushed to and automatically published from.

Latest version can be seen here:
https://mehta-lab.github.io/microDL

Thank you for giving me permissions to change the Pages settings in order to set this up. I appreciate your help and feedback.

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -6,6 +6,21 @@
...
+   :caption: Introduction:
+
+microDL is a deep learning framework for processing microscopy data.
+The framework consists of
+
+* Preprocessing: flatfield correction, mask creation and normalization
+* Training: Building and training 2.5D or 2D U-Net models for regression or segmentation
+* Inference: Use your trained model to predict full size images.
+
+This documentation goes in depth into the functions and classed of microDL.
```
Comment (ziw-liu) [inline docs/source/index.rst:20]: `classed` -> `classes`?

**Comment by jennyfolkesson** [comment]
This PR is ready for another review. I've tested it using the actions_docs.yml workflow seen here: 
https://github.com/mehta-lab/microDL/pull/194/commits/698c03aa27b3004ea4886c5c85194443619dfb04
and I'm assuming the behavior will be the same now that I've switched the branch that triggers the action to main. Thanks.

File: .github/workflows/actions_docs.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  push:
+    branches:
+      - main
+
+jobs:
+  docs:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
```
Comment (ziw-liu) [inline .github/workflows/actions_docs.yml:13]: Latest version for `actions/checkout` is v3, and `actions/setup-python` is v4. Is there a specific reason to use the older actions?

File: .github/workflows/actions_docs.yml [hunk 3]
```diff
@@ -23,7 +25,7 @@ jobs:
           make -C docs/ html
       - name: Deploy
         uses: peaceiris/actions-gh-pages@v3
-        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }}
+        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/actions_docs' }}
```
Comment (ziw-liu) [inline .github/workflows/actions_docs.yml]: Does this mean that we need to keep the `actions-docs` branch even after merging into `main`?

File: .github/workflows/actions_docs.yml (hunk 2 above)
Comment (jennyfolkesson) [reply .github/workflows/actions_docs.yml]: No reason, actions are now updated to latest versions.

File: .github/workflows/actions_docs.yml (hunk 3 above)
Comment (jennyfolkesson) [reply .github/workflows/actions_docs.yml]: No. I've just changed that line for testing with the current branch, it's reverted back to main again now.

**Review by ziw-liu — APPROVED** [review]

**Pull request merged**